expfmt/decode.go

```diff
@@ -107,7 +107,23 @@ type protoDecoder struct {
 // Decode implements the Decoder interface.
 func (d *protoDecoder) Decode(v *dto.MetricFamily) error {
 	_, err := pbutil.ReadDelimited(d.r, v)
-	return err
+	if err != nil {
+		return err
+	}
+	for _, m := range v.GetMetric() {
+		if m == nil {
+			continue
+		}
+		for _, l := range m.GetLabel() {
+			if l == nil {
+				continue
+			}
+			if !model.LabelNameRE.MatchString(l.GetName()) {
+				return fmt.Errorf("invalid label name %q", l.GetName())
+			}
```

Comment:
I'm 80% sure that protobuf can transport invalid UTF-8 strings (while JSON can't). So (just yesterday) I ran into a library function to check if a string is a valid UTF-8 string (on which we depend in our hashing function).

So I'd add
`if !utf8.ValidString(l.GetValue()) { ... } `